varstored-watch/varstored-watch.c

```diff
@@ -0,0 +1,186 @@
+#define _GNU_SOURCE
+#include <err.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <sys/types.h>
+#include <stdbool.h>
+#include <xenctrl.h>
+#include <getopt.h>
+#include <libgen.h>
+#include <syslog.h>
+
+static struct option varstored_watch_option[] = {
+    {"domid", 1, NULL, 'd'},
+    {"uuid", 1, NULL, 'u'},
+    {NULL, 0, NULL, 0}
+};
+
+static bool
+check_domain(domid_t domid)
+{
+    int rc;
+    xc_dominfo_t dominfo;
+    xc_interface *xch = NULL;
+
+    xch = xc_interface_open(NULL, NULL, 0);
+    if (!xch) {
+        syslog(LOG_ERR, "Failed to open xc_interface handle: %d, %s\n",
+            errno, strerror(errno));
+        goto err;
+    }
+
+    rc = xc_domain_getinfo(xch, domid, 1, &dominfo);
+    if (rc < 0) {
+        syslog(LOG_ERR, "Failed to get domain info: %d, %s\n", errno, strerror(errno));
+        goto err;
+    }
+    if (dominfo.domid != domid) {
+        syslog(LOG_ERR, "Domid %u does not match expected %u\n", dominfo.domid, domid);
+        goto err;
+    }
+    if (dominfo.shutdown == true) {
+        syslog(LOG_ERR, "Domid %u is shutting down.\n", domid);
```

Comment:
It'd be nicer to avoid having to open xc_interface to avoid the need for privcmd permissions.  Would checking /local/domain/$domid be sufficient as an existence check?  libxl will kill this process, so it doesn't need to exit itself on domain shutdown.